server/src/main/java/io/crate/execution/engine/collect/DocValuesAggregates.java

```diff
@@ -100,42 +100,37 @@
         ShardId shardId = indexShard.shardId();
         SharedShardContext shardContext = collectTask.sharedShardContexts().getOrCreateContext(shardId);
         Searcher searcher = shardContext.acquireSearcher(LuceneShardCollectorProvider.formatSource(phase));
-        try {
-            QueryShardContext queryShardContext = shardContext.indexService().newQueryShardContext();
-            collectTask.addSearcher(shardContext.readerId(), searcher);
-            LuceneQueryBuilder.Context queryContext = luceneQueryBuilder.convert(
-                phase.where(),
-                collectTask.txnCtx(),
-                indexShard.mapperService(),
-                indexShard.shardId().getIndexName(),
-                queryShardContext,
-                table,
-                shardContext.indexService().cache()
-            );
+        QueryShardContext queryShardContext = shardContext.indexService().newQueryShardContext();
```

Comment:
I think the `indexService()` call could fail, in which case the `searcher` wouldn't be released because it hasn't been added to the `collectTask` yet.

Makes me wonder if we should remove the lazy `indexService` initialization within `SharedShardContext` and do it eagerly. Then it couldn't fail here anymore.

Reply:
```
        Engine.Searcher searcher = sharedShardContext.acquireSearcher(formatSource(collectPhase));
        QueryShardContext queryShardContext = sharedShardContext.indexService().newQueryShardContext();
```
`acquireSearcher` preseeds to the `indexService()` and it calls `indexShard()` that call the `indexService()`, so if `indexService()` fails the searcher won't even be created, right?

Reply:
Ah you're right :+1: